### ISSUE TITLE
Add self-hosted macOS runner support?

### ISSUE THREAD

**Issue by mrmachine** [docker/setup-qemu-action#24]
### Behaviour

Docker Desktop works on macOS (Docker runs in a Linux VM managed by Docker Desktop).

GitHub Actions docs say only Linux runners are supported for Docker builds.

Multi-arch builds on GitHub hosted runners via `setup-qemu-action` are INSANELY slow (from 10 minutes to 1.5 hours). The same multi-arch builds on my Intel MacBook Pro with Docker Desktop is very fast.

It would be great if I could utilise my own hardware to build multi-arch images quickly with GitHub Actions.

#### Steps to reproduce this issue

Send a job using `setup-qemu-action` to a self hosted macOS runner.

#### Expected behaviour

`setup-qemu-action` should work on self hosted macOS runners.

#### Actual behaviour

When I attempt a build on a self-hosted macOS runner I get this error:

```
Error: Only supported on linux platform
```

### Configuration

```yml
name: ci

on:
  push:
    branches:
      - '*multiarch'

jobs:
  ci:
    runs-on:
      - macOS
      - self-hosted
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Docker meta
        id: docker_meta
        uses: crazy-max/ghaction-docker-meta@v1
        with:
          images: FOO/BAR
          tag-sha: true

      - name: Set up QEMU
        uses: docker/setup-qemu-action@v1

... snipped since error occurs in `setup-qemu-action`
```

### Logs

```
Run docker/setup-qemu-action@v1
  with:
    image: tonistiigi/binfmt:latest
    platforms: all
Error: Only supported on linux platform
```

**Comment by tonistiigi** [comment]
@mrmachine So if I understand correctly, you do have Docker installed in your macos self-hosted runner?

**Comment by mrmachine** [comment]
I run a GitHub self hosted runner in a container with docker desktop for Mac.

**Comment by tonistiigi** [comment]
> I run a GitHub self hosted runner in a container with docker desktop for Mac.

Why would it detect it as macos then if it is already in Linux container.

@crazy-max Anyway, I think we can allow this if `docker` is available.

**Comment by mrmachine** [comment]
Actually, sorry, it's been a while since I submitted this ticket.

I think I am NOW running GitHub Actions inside a container. However, I was wanting to run the GitHub Actions runner natively (per their recommended installation instructions). Since `docker` CLI is available on macOS, even if the engine is running inside a managed VM, I am not sure why the GitHub Actions runner cannot support Docker based actions.

As I said, ARM builds on hosted GitHub Actions runners are **insanely** slow. And the only option (I have found) for cloud based self hosted runners is to use an AWS t4g instance which is an unnecessary expense when I have an ARM based Mac that can run Docker and good bandwidth at home.
